lua/autorun/nl_library.lua

```diff
@@ -0,0 +1,476 @@
+	
+	local dirvec
+	local m = magn
+	local i = 1
+	local its = 1
+	repeat
+		dirvec = directions[i] * m
+		i = i + 1
+		if i > #directions then
+			its = its + 1
+			i = 1
+			m = m + magn
+			if its > iterations then
+				ply:SetPos(pos) // We've done as many checks as we wanted, lets just force him to get stuck then.
+				return false
+			end
+		end
+		
+		tr = NLL.PlayerHullTrace(dirvec + pos, ply, filter)
+	until tr.Hit == false
+	
+	ply:SetPos(pos + dirvec)
+	return true
+end
+
+
+/*
+Sets the player invisible/visible
+*/
+function NLL.PlayerInvis( ply, bool )
+	ply:SetNoDraw(bool)
+	ply:DrawShadow(not bool)
+
+	if pac and pac.TogglePartDrawing then
+		pac.TogglePartDrawing(ply, not bool)
+	end
+	/*
+	ply:SetMaterial( bool and "models/effects/vol_light001" or "" )
+	ply:SetRenderMode( bool and RENDERMODE_TRANSALPHA or RENDERMODE_NORMAL )
+	ply:Fire( "alpha", bool and 0 or 255, 0 )
+	*/
+end
+
+
+/*
+Deploy player ragdoll, Sleeping Variants can be woken up with E.
+*/
+function NLL.Ragdoll( ply, pushdir, stuntype)
+	if not SERVER then return end
+	local plyphys = ply:GetPhysicsObject()
+	local plyvel = Vector(0,0,0)
+	if plyphys:IsValid() then
+		plyvel = plyphys:GetVelocity()
+	end
+	
+	ply.nlldroppos = ply:GetPos() // Store pos incase the ragdoll is missing when we're to unrag him.
+	
+	local rag = ents.Create("prop_ragdoll")
+		rag:SetModel(ply:GetModel())
+		rag:SetPos(ply:GetPos())
+		rag:SetAngles(Angle(0,ply:GetAngles().y,0))
+		//FromPlyColor(ply:GetPlayerColor())) //Clothes get correct color, but the head gets pitchblack. :(
+		rag:SetColor(ply:GetColor())
+		rag:SetMaterial(ply:GetMaterial())
+		rag:Spawn()
+		rag:Activate()
+	
+	if not IsValid(rag:GetPhysicsObject()) then
+		MsgN("A slept player didn't get a valid ragdoll. Model ("..ply:GetModel()..")!")
+		SafeRemoveEntity(rag)
+		return false
+	end
+	
+	//Lower inertia makes the ragdoll have trouble rolling. Citizens have 1,1,1 as default, while combines have 0.2,0.2,0.2.
+	rag:GetPhysicsObject():SetInertia(Vector(1,1,1)) 
+	
+	//Push him back abit
+	plyvel = plyvel + pushdir*200
+	rag:GetPhysicsObject():SetVelocity(plyvel)
+		
+	//Stop firing of weapons
+	NLL.PlyStoreWeapons(ply)
+	ply:StripWeapons()
+	
+	//Makes him not collide with anything, including traces.
+	ply:SetCollisionGroup(COLLISION_GROUP_IN_VEHICLE)
+	
+	//Make him follow the ragdoll, if the player gets away from the ragdoll he won't get stuff rendered properly.
+	ply:SetParent(rag)
+	
+	//Make the player invisible.
+	NLL.PlayerInvis(ply, true)
+
+	ply.nllragdoll = rag
+	rag.nllplayer = ply
+	ply.condition = stuntype
+		
+	ply:SetNWEntity("nllviewrag", rag)
+	rag:SetNWEntity("nllplyowner", ply)
+	
+	ply.lasthp = ply:Health()
+	net.Start("nllsendhealth")
+		net.WriteEntity(ply)
+		net.WriteInt(ply:Health(),32)
+	net.Broadcast()
+	net.Start("nllplayercond")
+	net.WriteEntity(ply)
+	net.WriteString(stuntype)
+	net.Broadcast()
+	
+	return true
+end
+
+function NLL.UnRagdoll( ply )
+	if not SERVER then return end
```

Comment:
If this function is only meant for server, then only define it there using `if SERVER then`

Reply:
`if not SERVER then return end` line is still there